### PR TITLE
[TechDocs] Create Text Size Addon

### PR DESCRIPTION
**Pull request by camilaibs**
## Hey, I just made a Pull Request!

This pull request creates a new TechDocs Addon that gives users the ability to increase or decrease the font size on documentation pages.

Initially, this setting is being kept in the browser's local storage and the font size range is not customizable for now but could be in the future if needed.

For this addon to work, I also added a new addons location, which is "Settings" and this location is a menu located in the subheader of the reader page, designed to contain settings that allow readers to customize their reading experience, such as accessibility.

text-size-addon

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/techdocs-module-addons-contrib/api-report.md [hunk 1]
```diff
@@ -31,4 +31,7 @@ export type ReportIssueTemplateBuilder = ({
 
 // @public
 export const techdocsModuleAddonsContribPlugin: BackstagePlugin<{}, {}>;
+
+// @public
+export const TextSize: (props: unknown) => JSX.Element | null;
```
Comment (camilaibs) [inline plugins/techdocs-module-addons-contrib/api-report.md:36]: Unknown type will be fixed here:
https://github.com/backstage/backstage/pull/11166/files#diff-03813941362a23884f6471cbbb50dd7cb7058cfe5cacabfff4e2612619610ec7

File: packages/app/src/App.tsx [hunk 2]
```diff
@@ -183,6 +186,7 @@ const routes = (
       {techDocsPage}
       <TechDocsAddons>
         <ReportIssue />
+        <TextSize />
```
Comment (camilaibs) [inline packages/app/src/App.tsx:189]: Should I add it to the embedded app in the TechDocs CLI?

Comment (iamEAP) [reply packages/app/src/App.tsx:189]: Hmm... Either way!

Would be nice to figure out a way for CLI users to pick which are enabled/disabled... But that's a rabbit hole for another time.

**Review by iamEAP — APPROVED** [review]
Love everything about this: the new addon location! The docs! The functionality!

Nice work.

File: packages/app/src/App.tsx (hunk 2 above)
Comment (camilaibs) [reply packages/app/src/App.tsx:189]: Agree, thanks!

**Review by realandersn — APPROVED** [review]
Nice! 🚀

**Pull request merged**